### PR TITLE
fix(mlir): while and cond computation edge cases

### PR DESCRIPTION
**Pull request by polvalente**
- fix(mlir): while and cond edge cases
- chore: unskip more tests

### CODE REVIEW

File: exla/lib/exla/defn.ex [hunk 1]
```diff
@@ -2048,9 +2048,19 @@ defmodule EXLA.Defn do
...
     [arg_token | arg_params] = EXLA.MLIR.Function.get_arguments(function)
 
-    arg_params = Value.tuple(arg_params)
+    params =
+      case arg_params do
+        _ when is_tuple(arg) ->
+          {arg, Value.tuple(arg_params)}
+
+        _ when is_list(arg) ->
+          arg |> Enum.zip(arg_params) |> List.to_tuple()
```
[the remark this answers is not in the document]
Comment (polvalente) [reply exla/lib/exla/defn.ex:2057]: I think `is_list` is superfluous here. I'm checking, but this might just become a `if is_tuple` statement.

[the remark this answers is not in the document]
Comment (polvalente) [reply exla/lib/exla/defn.ex]: Indeed, we just need to branch on whether the argument is a tuple or not.

the MLIR interface doesn't accept tuples as arguments, so we need to flatten it when building the function, basically

**Review by josevalim — APPROVED** [review]
